src/function/open-referral-function/src/FamilyHubs.OpenReferral.Function/ClientServices/HsdaApiService.cs

```diff
@@ -0,0 +1,66 @@
+using System.Net;
+using FamilyHubs.OpenReferral.Function.Entities;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace FamilyHubs.OpenReferral.Function.ClientServices;
+
+public class HsdaApiService(ILogger<HsdaApiService> logger, HttpClient httpClient) : IHsdaApiService
+{
+    public async Task<(HttpStatusCode, JArray?)> GetServices()
+    {
+        (HttpStatusCode httpStatusCode, string? jsonResponse) = await HttpGet("/services");
+
+        if (httpStatusCode != HttpStatusCode.OK || jsonResponse is null)
+        {
+            logger.LogError(
+                "Failed to get the service list from /services | Status Code = {httpStatusCode}, JSON = {jsonResponse}",
+                httpStatusCode, jsonResponse);
+            return (httpStatusCode, null);
+        }
+
+        JArray serviceList = JArray.Parse(JObject.Parse(jsonResponse)["contents"]!.ToString());
+
+        if (serviceList.Count == 0)
+        {
+            logger.LogInformation("Query was OK, but no services were found");
+            return (HttpStatusCode.NotFound, null);
```

Comment:
Not sure if a 404 NotFound is the best status code here, as it should really only be used if a specific resource isn't found. Since we're really asking for a list of services, and in this case none are returned, we could potentially use 204 NoContent, indicating a success, but no associated content; 404 is more of a failure response.

Reply:
I can change to that, mostly stuck with 404 because the 2nd AC feels a little ambiguous and wasn't sure how exact I had to be

Reply:
Changed in: [d515f73](https://github.com/DFE-Digital/fh-services/pull/63/commits/d515f733e5a66a8ca89d7f0fae5cf0d539b7c730)

(Also added a missed test for that condition)